### PR TITLE
Adds Toggle Button and Menu components.

### PR DESCRIPTION
**Pull request by sbartlett**

### CODE REVIEW

File: src/components/containers/menu/Menu.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  state = {
+    isMenuOpen: false
+  };
+
+  getChildContext() {
+    return { inline: this.props.inline };
+  }
+
+  toggleMenu = () => {
+    this.setState({
```
Comment (jrios) [inline src/components/containers/menu/Menu.js]: When you use `setState` and are relying on previous state, you should use the function argument method instead:

```javascript
this.setState(previousState => ({ isMenuOpen: !previousState.isMenuOpen }))
```

The reason for this is that state changes are queued and not guaranteed to be in order. Although it's unlikely to be a problem in this scenario, it's safer to use the function syntax.

File: src/components/controls/buttons/ToggleButton.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import defaultTheme from '../../theme/defaultTheme';
+import { buttonStyleTypes, defaultButtonVariants } from './button-variants';
+
+class ToggleButton extends React.Component {
+  state = {
+    isPressed: false
+  };
+
+  toggleButton = event => {
+    this.setState({ isPressed: !this.state.isPressed });
```
Comment (jrios) [inline src/components/controls/buttons/ToggleButton.js]: Same as my comment above.

File: src/index.js [hunk 3]
```diff
@@ -6,6 +6,8 @@ export Fieldset from './components/containers/fieldset/Fieldset';
 export Tooltip from './components/containers/tooltip/Tooltip';
 export PopoverLink from './components/containers/popover/PopoverLink';
 export Modal from './components/containers/modal/Modal';
+export Menu from './components/containers/menu/Menu';
+export MenuSection from './components/containers/menu/MenuSection';
```
Comment (jrios) [inline src/index.js]: This doesn't need to be exported. It is exposed on the `Menu` component.

Also missing the `ToggleButton` component here.

File: src/components/navigation/breadcrumb/Breadcrumb.js [hunk 4]
```diff
@@ -1,23 +1,23 @@
...
-  );
-}
-
-Breadcrumb.propTypes = {
-  keySelector: PropTypes.func.isRequired,
-  children: PropTypes.node.isRequired
-};
-
-export default Breadcrumb;
+import React from 'react';
```
Comment (jrios) [inline src/components/navigation/breadcrumb/Breadcrumb.js]: This shouldn't be included in the PR.

File: src/components/base/icons/icon-definitions.js [hunk 5]
```diff
@@ -1,163 +1,160 @@
...
-  ['oe-icon-lt-arrow-left']: '60e',
-  ['oe-icon-lt-arrow-right']: '60f',
-  ['oe-icon-lt-arrow-up']: '610'
-};
-
-export {
-  regular,
-  light
-};
+const regular = {
```
Comment (jrios) [inline src/components/base/icons/icon-definitions.js]: Is this needed?

File: src/components/containers/menu/MenuPanel.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        </StyledChildrenContainer>
+      </StyledPanel>
+    </ThemeProvider>
+  );
+};
+
+MenuPanel.propTypes = {
+  children: PropTypes.any.isRequired,
+  isOpen: PropTypes.bool.isRequired,
+  closeFunction: PropTypes.func.isRequired
```
Comment (jrios) [inline src/components/containers/menu/MenuPanel.js]: Is there a requirement that a `closeFunction` is always provided?

File: src/components/containers/menu/MenuPanel.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+          {children}
+        </StyledChildrenContainer>
+      </StyledPanel>
+    </ThemeProvider>
+  );
+};
+
+MenuPanel.propTypes = {
+  children: PropTypes.any.isRequired,
+  isOpen: PropTypes.bool.isRequired,
```
Comment (jrios) [inline src/components/containers/menu/MenuPanel.js]: Could this be an optional prop with a default value?

File: src/components/containers/menu/Menu.js (hunk 1 above)
Comment (sbartlett) [reply src/components/containers/menu/Menu.js]: Thanks, didn't know that.
